### ISSUE TITLE
Register with shouldUnregister behaves unconsistently

### ISSUE THREAD

**Issue by stropitek** [react-hook-form/react-hook-form#8011]
### Version Number

7.27.1

### Codesandbox/Expo snack

https://codesandbox.io/s/wandering-thunder-95ideu?file=/src/App.js

### Steps to reproduce

1. Put cursor in first input
2. Enter a value in the input
3. Click remove to unmount input
4. Click add to remount input
5. See how the input is now empty because of shouldUnregister
6. Repeat 2., 3. and 4. for the second input
7. See how the input still has the entered value
8. Repeat 2., 3. and 4. for the third input
10. See how the input still has the entered value

### Expected behaviour

The second and third inputs should behave the same way the first input behaves. The 2nd and 3rd inputs behave as if `shouldUnregister` was not specified.

### What browsers are you seeing the problem on?

Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
please refer to the doc under rules.

https://react-hook-form.com/api/useform#shouldUnregister

> unmounted input will need to notify at either useForm, or useWatch's useEffect for hook form to verify input is unmounted from the DOM.

you can also manually invoke [unregister](https://react-hook-form.com/api/useform/unregister) method

**Comment by stropitek** [comment]
Thank you, that helps!

**Comment by stropitek** [comment]
It seems easy to make a mistake with this option. Do you think it would be technically possible to throw an error when the api is not used correctly?

**Comment by bluebill1049** [comment]
I don't think so. If this is a concern I would suggest using the default option `shouldUnregister: false` and always `unreigster` at the application usage level.

**Comment by stropitek** [comment]
Yes indeed, this is the way I'll use it from now on, thanks again
